packages/keycodes/src/test/index.js

```diff
@@ -230,3 +231,273 @@ describe( 'rawShortcut', () => {
 		} );
 	} );
 } );
+
+describe( 'isKeyboardEvent', () => {
+	afterEach( () => {
+		while ( document.body.firstChild ) {
+			document.body.removeChild( document.body.firstChild );
+		}
+	} );
+
+	function keyPress( target, modifiers = {} ) {
+		[ 'keydown', 'keypress', 'keyup' ].forEach( ( eventName ) => {
+			const event = new window.Event( eventName, { bubbles: true } );
+			Object.assign( event, modifiers );
+			target.dispatchEvent( event );
+		} );
+	}
+
+	function attachEventListeners( target, eventHandler ) {
+		[ 'keydown', 'keypress', 'keyup' ].forEach( ( eventName ) => {
+			target.addEventListener( eventName, eventHandler );
+		} );
+	}
+
+	describe( 'primary', () => {
+		it( 'should identify modifier key when Ctrl is pressed', ( done ) => {
+			const attachNode = document.createElement( 'div' );
+			document.body.appendChild( attachNode );
+			attachEventListeners( attachNode, ( event ) => {
+				expect( isKeyboardEvent.primary( event, undefined, isAppleOSFalse ) ).toBe( true );
+				done();
```

Comment:
`keyPress` triggers 3 event types at once. Will `attachEventListeners` be able to process all 3 verifications? It might finish on the first verification just after `done` is called with how this is coded at the moment.

Reply:
@gziolo Yes i believe you are right, I'll fix this. On the same point id like to clarify how such a test should be executed.

Each key event (Keydown, Keyup and Keypress) should be ideally tested for each of the tests. Following a one assert per test case policy will soon make this test very large. The internet seems to be divided on what is the correct practice for this with some arguing that a test should check the functionality and not the result of a functionality. 

I just wanted to know the flow expected by the Gutenberg team

Reply:
> Each key event (Keydown, Keyup and Keypress) should be ideally tested for each of the tests. Following a one assert per test case policy will soon make this test very large.

I don't think we use [describe.each(table)(name, fn, timeout)](https://jestjs.io/docs/en/api#describeeachtable-name-fn-timeout) in other test files but it sounds like a great opportunity to give it a try.

> Use `describe.each` if you keep duplicating the same test suites with different data. describe.each allows you to write the test suite once and pass data in.

Reply:
> I just wanted to know the flow expected by the Gutenberg team.
> 
> There also seem to be two tests in the repo that do assert more than once per test. Since the number of these seemed small, I just wanted to reconfirm.

We have over 300 test suites so I'm sure it varies depending on who coded and reviewed it :)

Reply:
> > Each key event (Keydown, Keyup and Keypress) should be ideally tested for each of the tests. Following a one assert per test case policy will soon make this test very large.
> 
> I don't think we use [describe.each(table)(name, fn, timeout)](https://jestjs.io/docs/en/api#describeeachtable-name-fn-timeout) in other test files but it sounds like a great opportunity to give it a try.
> 
> > Use `describe.each` if you keep duplicating the same test suites with different data. describe.each allows you to write the test suite once and pass data in.

`describe.each` will still log each case independently while making the test case code more readable. I was considering using something like `expect.assertions(3);` which will still assert all 3 cases but log only a single line indicating that all 3 tests have passed. What do you think of this approach? Or is such a high level of granularity in the test output preferred?

Reply:
Yes, it might be another solution if you think that it's going to be easier to follow for others this way. You can decide.